### DISCUSSION TITLE
Can 2 verdaccio servers fetch packages from each other?

### DISCUSSION THREAD

**Discussion by sepsol** [verdaccio/verdaccio#2721] [Q&A]
So I have a local verdaccio server running inside a virtual machine on my computer and connected via bridge to my LAN. My colleagues and I do our development on this server locally. On the other hand, whenever we want to deploy our project we would temporarily expose our package server over the internet using `ngrok` and then we'll install any necessary internal packages on our production virtual machine and deploy the project.

Now, we're trying to make this flow smoother by creating a second instance of our main package server which simply acts as a cache for our production environment. We want to be able to manually expose our local instance and go to our fallback instance and ask it to fetch whatever packages are available on the local server to there so that our production environment has access to all the packages.

The reason we want to do this is so that we can have access to the `npx` command in our packages in the production environment without exposing our registry over the internet.

Are there any solutions to this issue as of now? I've been searching in the github issues, discussions, documentation, etc. but was not able to find anything useful.

**Comment by sepsol** [comment]
To clarify, I want the second remote package server to act as a mirror (and not a proxy) to the main local package server.

**Reply by juanpicado** [reply]
hi @sepsol I've been thinking a bit these days about your question, the registry can fetch by demand, but they cannot sync each other unless you do so, but assuming you have the default storage, have you considered just copying the storage to the second location? I am just thinking louder.

**Reply by sepsol** [reply]
I've thought of copying over the contents of the storage folder but was not sure if that's the best way and how I should do it over the internet. The best way I could think of was to track it inside a git repository and pull it from Github on the remote server but again, that seems to add redundancy which I don't know if I'd like. 

Can you guide me in the right direction for how I should fetch the contents of the storage folder on-demand? I think if implemented correctly, this is the best way for me since I don't want to expose my local server indefinitely over the internet.

**Reply by juanpicado** [reply]
I have no much idea now, but just my .5 cents, could work via SSH with SCP?

**Reply by sepsol** [reply]
Thanks @juanpicado, I'll look into it and update this thread with the results whenever I get a chance.
